datafusion-examples/examples/composed_extension_codec.rs

```diff
@@ -266,26 +314,35 @@ impl PhysicalExtensionCodec for ComposedPhysicalExtensionCodec {
         inputs: &[Arc<dyn ExecutionPlan>],
         registry: &dyn FunctionRegistry,
     ) -> Result<Arc<dyn ExecutionPlan>> {
-        self.try_any(|codec| codec.try_decode(buf, inputs, registry))
+        let (codec, blob) = self.decode_protobuf(buf)?;
```

Comment:
we have repeated code in multiple functions should we consider merging those two functions?